assets/src/edit-story/components/canvas/frameElement.js

```diff
@@ -65,6 +65,13 @@ function FrameElement({ element }) {
   const { id, type } = element;
   const { Frame, isMaskable, Controls } = getDefinitionForType(type);
   const elementRef = useRef();
+  const [hovering, setHovering] = useState(false);
+  const {
+    state: { isAnythingTransforming },
+  } = useTransform();
+
+  const onPointerEnter = () => setHovering(true);
```

Comment:
Out of curiosity: Why do want it here? Is it future proofing in case we want `React.memo` the whole Wrapper? or we count on some performance gains? - [that will not work as expected](https://kentcdodds.com/blog/usememo-and-usecallback).

Reply:
Yeah, in this case you can probably skip. Though it'd be better to inline I think, otherwise it's just what I often react to. Otherwise, it's really only matters how deep this callback propagates, and in this case it doesn't go very deep.

Reply:
But even if it goes N levels deep (in component tree), if we don't utilize `React.memo` at some point - there is no benefit (not talking about valid use cases when stable identity is required [most of the time on state/provider level] or memo usage in components like Library).

JSX can be already convoluted by conditions, so for readability sake I prefer to not inline them.

```jsx
<Wrapper
        ref={elementRef}
        data-element-id={id}
        {...box}
        onMouseDown={(evt) => {
          if (isSelected) {
            elementRef.current.focus({ preventScroll: true });
          } else {
            handleSelectElement(id, evt);
          }
          if (!isBackground) {
            evt.stopPropagation();
          }
        }}
        onFocus={(evt) => {
          if (!isSelected) {
            handleSelectElement(id, evt);
          }
        }}
        onPointerEnter={onPointerEnter}
        onPointerLeave={onPointerLeave}
        tabIndex="0"
        aria-labelledby={`layer-${id}`}
        hasMask={isMaskable}
        data-testid="frameElement"
      >
        <WithMask element={element} fill={true}>
          {Frame ? (
            <Frame wrapperRef={elementRef} element={element} box={box} />
          ) : (
            <EmptyFrame />
          )}
        </WithMask>
      </Wrapper>
```
vs
```jsx
<Wrapper
        ref={elementRef}
        data-element-id={id}
        {...box}
        onMouseDown={onMouseDown}
        onFocus={onFocus}
        onPointerEnter={onPointerEnter}
        onPointerLeave={onPointerLeave}
        tabIndex="0"
        aria-labelledby={`layer-${id}`}
        hasMask={isMaskable}
        data-testid="frameElement"
      >
        <WithMask element={element} fill={true}>
          {Frame ? (
            <Frame wrapperRef={elementRef} element={element} box={box} />
          ) : (
            <EmptyFrame />
          )}
        </WithMask>
      </Wrapper>
```

Reply:
I meant more where the callbacks end up used as deps for `useEffect` hooks deeper downstream and start triggering extra hook invocations. But your point is well taken. This is a safer case so feel free to resolve and leave unchanged.